src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,12 +8,98 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, some street, #12-34, 123456";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Format: blockNumber, street, unitNumber, postalCode";
+    public static final String ADDRESS_VALIDATION_REGEX = "(\\d+),\\s+([a-zA-Z0-9\\s]+),\\s+(#\\d+-\\d+),\\s+(\\d+)";
 
-    public final String value;
     private boolean isPrivate;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
+
+    /**
+     * Represents a housing Block for a person in the address book
+     */
+    private class Block {
+
```

Comment:
You might also try declaring these new classes as stand-alone classes and ensuring compositionality using private instantiation within Address class. Please read this link for more information: https://www.journaldev.com/1325/composition-in-java-example